### PR TITLE
Feature generate qrcode

### PR DESCRIPTION
**Pull request by Walorda**
Android components should be placed in android files ---- is broken

### CODE REVIEW

File: src/screens/qrcode_generator/Test.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    private qr: React.RefObject<QrcodeGenerator>;
+    constructor(){
+        super({});
+        this.qr = React.createRef();
+    }
+
+    /*
+     *         <Button title="test" onPress={() => this.qr.current?.requestCameraPermission}/>
+     *           This is how you would use it.
+     *  ADDED IT IN COMMENTS BECAUSE FUCKING ESLINT + REACT NATIVE + TYPESCRIPT IS A CANCEROUS FUCKING TUMOR
```
Comment (mvegter) [inline src/screens/qrcode_generator/Test.tsx]: ...

File: src/navigations/Navigation.tsx [hunk 2]
```diff
@@ -17,6 +17,7 @@ const RootNavigator = createSwitchNavigator(
         Recipe: RecipeNavigator,
         Profile:ProfileScreen,
         SettingScreen: SettingsScreen
+
```
Comment (mvegter) [inline src/navigations/Navigation.tsx]: ?

File: src/screens/qrcode_generator/QrcodeGenerator.ios.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+import React from 'react';
+import CameraRoll from '@react-native-community/cameraroll';
+import RNFS from 'react-native-fs';
+
+interface QrProps {
+  value: string;
+}
+/**
+ * Component for generating an qr code.
+ * Make the component invissble or hidden and call the fucking requestcamera which in turns makes the qr code when
```
Comment (mvegter) [inline src/screens/qrcode_generator/QrcodeGenerator.ios.tsx]: ...

File: src/screens/qrcode_generator/QrcodeGenerator.android.tsx [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+import CameraRoll from '@react-native-community/cameraroll';
+import RNFS from 'react-native-fs';
+import { PermissionsAndroid } from 'react-native';
+
+interface QrProps {
+  value: string;
+}
+/**
+ * Component for generating an qr code.
+ * Make the component invissble or hidden and call the fucking requestcamera which in turns makes the qr code when
```
Comment (mvegter) [inline src/screens/qrcode_generator/QrcodeGenerator.android.tsx]: ...

**Review by mvegter — CHANGES_REQUESTED** [review]

File: src/screens/qrcode_generator/QrcodeGenerator.android.tsx [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+  public async requestCameraPermission() {
+      try {
+          const granted = await PermissionsAndroid.request(
+              PermissionsAndroid.PERMISSIONS.WRITE_EXTERNAL_STORAGE,
+              {
+                  title: 'Permission to store image',
+                  message:
+            'Permission to store image',
+                  buttonNeutral: 'Ask Me Later',
```
Comment (Jordy-Boelhouwer) [inline src/screens/qrcode_generator/QrcodeGenerator.android.tsx]: Is the neutral option neccesary?

File: src/screens/qrcode_generator/QrcodeGenerator.ios.tsx [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+  public saveQrToDisk() {
+      this.svg.toDataURL((data: any) => {
+          const date = new Date().toLocaleDateString().split('/').join('-');
+          RNFS.writeFile(RNFS.DocumentDirectoryPath + '/' + date + '.png', data, 'base64')
+              .then(() => {
+                  return CameraRoll.saveToCameraRoll(RNFS.DocumentDirectoryPath + '/' + date + '.png', 'photo');
+              })
+              .then(() => {
+                  console.log('test');
```
Comment (Jordy-Boelhouwer) [inline src/screens/qrcode_generator/QrcodeGenerator.ios.tsx]: Remove the logs if you are done testing

File: src/screens/qrcode_generator/QrcodeGenerator.android.tsx (hunk 5 above)
Comment (Jordy-Boelhouwer) [inline src/screens/qrcode_generator/QrcodeGenerator.android.tsx]: Is the neutral option neccesary?

**Review by Jordy-Boelhouwer — CHANGES_REQUESTED** [review]

**Comment by mvegter** [comment]
@Walorda ping

**Comment by mvegter** [comment]
Ping @Walorda

@Jordy-Boelhouwer Can I get your opinion regarding the (error) eslint issues.

**Comment by mvegter** [comment]
@Walorda @Jordy-Boelhouwer does the `barcode_scanner` match the naming convention?

**Review by mvegter — APPROVED** [review]

**Pull request merged**